l1-contracts/src/core/Rollup.sol

```diff
@@ -142,8 +150,12 @@ contract Rollup is Leonidas, IRollup, ITestRollup {
     override(ITestRollup)
     onlyOwner
   {
-    if (blockNumber > provenBlockCount && blockNumber <= pendingBlockCount) {
-      provenBlockCount = blockNumber;
+    if (
+      blockNumber > (state.provenTip.blockNumber + 1)
```

Comment:
Some of these bounds seems weird to me. If you have a pending that is proven + 1, would you not just be failing all the time.

Reply:
Agreed, they are insane. I think we should purge this function/concept altogether.